AirCasting/CoreData/MeasurementStreamStorage.swift

```diff
@@ -101,6 +103,27 @@ final class CoreDataMeasurementStreamStorage: MeasurementStreamStorage {
         try context.save()
     }
 
+    func updateSessionIfFollowing(_ sessionFollowing: SessionFollowing, for sessionUUID: SessionUUID) throws {
+        let context = persistenceController.editContext()
+        let sessionEntity = try context.existingSession(uuid: sessionUUID)
+        let copySession = sessionEntity
+        if sessionFollowing.rawValue == 1 {
+            sessionEntity.followedAt = Date()
```

Comment:
all this should also be inside `performBlockAndWait`. Any reads/writes for CoreData-managed classes (`NSManagedObject` subclasses) should be wrapped with it.

Reply:
for this "please try without it and see if it changes anything. I'd be very surprised if it did. Reason is that SessionEntity is a class, so a reference type (unlike structs which are value types, see this), assigning does not make a copy of it." @pawelgil the copy needs to be made to make it appear on following like as on the fixed at the same time. Tested it now